custom/enikshay/tasks.py

```diff
@@ -147,11 +195,14 @@ def update_single_case(self, episode_case):
         if update_json:
             update_case(self.domain, episode_case.case_id, update_json)
 
-    def _get_open_episode_cases(self):
+    def _get_case_ids(self):
         case_accessor = CaseAccessors(self.domain)
         case_ids = case_accessor.get_open_case_ids_in_domain_by_type(CASE_TYPE_EPISODE)
```

Comment:
I did something like this https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/userreports/management/commands/async_rebuild_table.py#L49-L62 I think you would want to pass in one partition per task and set those tasks up sequentially. 

You can still get the related case from the other partition, but the main benefit is that this would split up the task in multiple different tasks. If this task is failing because of OOM then I think this might solve it since any memory leaks would disappear once the worker is restarted. Also if it's just a couple of cases in one partition causing issues it might help you to narrow it down